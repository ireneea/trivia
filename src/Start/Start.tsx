import React from "react";
import { View, StyleSheet } from "react-native";

import Button from "../components/Button";
import BackgroundScreen from "../components/BackgroundScreen";

import { getCountryQuizGame } from "../generateQuiz/countryQuiz";
import { filterCountries } from "../generateQuiz/filterCountries";
import allCountries from "../../data/countries.json";

type Props = {
  navigation: any;
};

const Start: React.FC<Props> = (props) => {
  const onStart = () => {
    const { navigation } = props;
    const validCountries = filterCountries(allCountries, { withCapital: true });
    const game = getCountryQuizGame(validCountries);

    navigation.navigate("Game", { game });
  };

  return (
    <BackgroundScreen testID="startScreen">
      <View style={styles.btnContainer}>
        <Button testID="startBtn" onPress={onStart} label="Start" />
      </View>
    </BackgroundScreen>
  );
};

const styles = StyleSheet.create({
  btnContainer: {
    paddingHorizontal: 20,
    width: "100%",
  },
});

export default Start;
