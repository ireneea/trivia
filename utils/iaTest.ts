import { ReactTestInstance } from "react-test-renderer";

export function findByTestID(
  root: ReactTestInstance,
  testID: string
): ReactTestInstance {
  return root.findByProps({ testID });
}

export function mockNavigationProps(props = {}) {
  return {
    navigation: {
      // when use in a unit test jest will be defined
      navigate: jest.fn(),
    },
    ...props,
  };
}
