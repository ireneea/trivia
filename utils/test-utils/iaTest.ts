export function mockNavigationProps(props: Object = {}): Object {
  return {
    navigation: {
      // when use in a unit test jest will be defined
      navigate: jest.fn(),
    },
    ...props,
  };
}

export function mockRouteParamsProps(params: Object, props: Object = {}): Object {
  return {
    route: {
      params,
    },
    ...props,
  };
}
