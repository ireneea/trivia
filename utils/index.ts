import * as test from "./iaTest";

export default {
  test,
};
