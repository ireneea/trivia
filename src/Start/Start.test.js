import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import Start from "./Start";
import { findByTestID, mockNavigationProps } from "../../utils/iaTest";

describe("Start", () => {
  it("should render correctly", () => {
    const { baseElement } = render(<Start />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should navigate to the Game page", () => {
    const props = mockNavigationProps();

    const { getByText } = render(<Start {...props} />);
    fireEvent.press(getByText("Start"));

    expect(props.navigation.navigate).toHaveBeenCalledWith("Game");
  });
});
