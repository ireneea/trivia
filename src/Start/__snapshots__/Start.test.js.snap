// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#56AFEA",
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="startScreen"
    >
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "width": "100%",
          }
        }
      >
        <TouchableWithoutFeedback
          testID="startBtn"
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4284702066,
                  4282728784,
                ],
                "endPoint": undefined,
                "locations": undefined,
                "startPoint": undefined,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 20,
                "marginVertical": 5,
                "paddingHorizontal": 50,
                "paddingVertical": 15,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                  "textTransform": "uppercase",
                }
              }
            >
              Start
            </Text>
          </ViewManagerAdapter_ExpoLinearGradient>
        </TouchableWithoutFeedback>
      </View>
    </View>
  </View>
</View>
`;
